contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/HDF5Utils.java

```diff
@@ -0,0 +1,254 @@
+      } else {
+        return new HDF5Attribute(MinorType.BIGINT, key, reader.uint64().getAttr(pathName, key));
+      }
+    } else if (type.isAssignableFrom(int.class)) {
+      if (attributeInfo.isSigned()) {
+        return new HDF5Attribute(MinorType.INT, key, reader.int32().getAttr(pathName, key));
+      } else {
+        return new HDF5Attribute(MinorType.INT, key, reader.uint32().getAttr(pathName, key));
+      }
+    } else if (type.isAssignableFrom(short.class)) {
+      if (attributeInfo.isSigned()) {
+        return new HDF5Attribute(MinorType.INT, key, reader.int16().getAttr(pathName, key));
+      } else {
+        return new HDF5Attribute(MinorType.INT, key, reader.uint16().getAttr(pathName, key));
+      }
+    } else if (type.isAssignableFrom(byte.class)) {
+      if (attributeInfo.isSigned()) {
+        return new HDF5Attribute(MinorType.INT, key, reader.int8().getAttr(pathName, key));
+      } else {
+        return new HDF5Attribute(MinorType.INT, key, reader.uint8().getAttr(pathName, key));
+      }
+    } else if (type.isAssignableFrom(double.class)) {
+      return new HDF5Attribute(MinorType.FLOAT8, key, reader.float64().getAttr(pathName, key));
+    } else if (type.isAssignableFrom(float.class)) {
+      return new HDF5Attribute(MinorType.FLOAT4, key, reader.float32().getAttr(pathName, key));
+    } else if (type.isAssignableFrom(String.class)) {
+      return new HDF5Attribute(MinorType.VARCHAR, key, reader.string().getAttr(pathName, key));
+    } else if (type.isAssignableFrom(boolean.class)) {
+      return new HDF5Attribute(MinorType.BIT, key, reader.bool().getAttr(pathName, key));
+    } else if (type.isAssignableFrom(HDF5EnumerationValue.class)) {
+      // Convert HDF5 Enum to String
+      return new HDF5Attribute(MinorType.GENERIC_OBJECT, key, reader.enums().getAttr(pathName, key));
+    } else if (type.isAssignableFrom(BitSet.class)) {
+      return new HDF5Attribute(MinorType.BIT, key, reader.bool().getAttr(pathName, key));
+    }
+
+    logger.info("Reading attributes of type " + attributeInfo + " not yet implemented.");
+    return null;
+  }
+
+  public static MinorType getDataType(final HDF5DataSetInformation datasetInfo) {
+
+    HDF5DataTypeInformation typeInfo = datasetInfo.getTypeInformation();
+    Class<?> type = typeInfo.tryGetJavaType();
+    String name = typeInfo.getDataClass().name();
+    if (type == null) {
+      logger.info("Datasets of type " + typeInfo + " not implemented.");
+      //Fall back to string
+      if(name.equalsIgnoreCase("OTHER")) {
+        return MinorType.GENERIC_OBJECT;
+      } else {
+        return MinorType.VARCHAR;
+      }
+    } else if (type.isAssignableFrom(long.class)) {
+      return MinorType.BIGINT;
+    } else if (type.isAssignableFrom(short.class)) {
+      return MinorType.SMALLINT;
+    } else if (type.isAssignableFrom(byte.class)) {
+      return MinorType.TINYINT;
+    } else if (type.isAssignableFrom(int.class)) {
```

Comment:
Does this work? Int is assignable from short and byte, so would be caught above? Should this work from widest to narrowest to handle such cases?

Do the tests specifically test the desired mappings to catch bugs like this?

Reply:
Shockingly, this does work.  There are 37 unit tests, and I tried to have at least 1 unit test for every data type, both single dimension and multidimensional.